pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -628,6 +631,51 @@ type (
 	}
 )
 
+// Git groups all configuration options related to enabling GitOps on a
+// cluster and linking it to a Git repository.
+type Git struct {
+	Repo *Repo `json:"repo,omitempty"`
+	// +optional
+	Operator *Operator `json:"operator,omitempty"`
+	// +optional
+	Profiles []*Profile `json:"profiles,omitempty"` // one or many profiles to enable on this cluster
+}
+
+// Repo groups all configuration options related to a Git repository used for
+// GitOps.
+type Repo struct {
+	URL string `json:"url,omitempty"` // the Git SSH URL to the repository which will contain the cluster configuration, e.g. git@github.com:org/repo
+	// +optional
+	Branch string `json:"branch,omitempty"` // the branch under which cluster configuration files will be committed & pushed, e.g. master
+	// +optional
+	Paths []string `json:"paths,omitempty"` // relative paths within the Git repository which the GitOps operator will monitor to find Kubernetes manifests to apply, e.g. ["kube-system", "base"]
+	// +optional
+	FluxDir string `json:"fluxDir,omitempty"` // the directory under which Flux configuration files will be written, e.g. flux/
+	User    string `json:"user,omitempty"`    // Git user which will be used to commit changes
```

Comment:
I think the user, technically speaking, was optional

Reply:
`Email` as well then, right?

Reply:
No, I think the email was marked as required. But I had an idea on how to make it optional and read it so I would say it's required for now and I'll try to make it optional

Reply:
Done.